std-bits/table/src/main/java/org/enso/table/data/column/storage/SpecializedStorage.java

```diff
@@ -139,13 +151,15 @@ public SpecializedStorage<T> slice(int offset, int limit) {
 
   @Override
   public SpecializedStorage<T> slice(List<SliceRange> ranges) {
+    Context context = Context.getCurrent();
     int newSize = SliceRange.totalLength(ranges);
     T[] newData = newUnderlyingArray(newSize);
     int offset = 0;
     for (SliceRange range : ranges) {
       int length = range.end() - range.start();
       System.arraycopy(data, range.start(), newData, offset, length);
```

Comment:
Actually, the amount of `ranges` will likely not be that big in comparison to their `totalLength`.

So to make this method interruptible, we should actually need to be able to interrupt `System.arraycopy` while its operating.

This pattern repeats in _many_ places throughought our Storage implementation as it tries to use `arraycopy` etc. to be efficient.

To make it interruptible we would have to create our own `interruptibleCopy` which would essentially be just a `for` loop copying from one array to another, hitting the `safepoint()` on each iteration.

However, I'm slightly worried that it may bring down the performance - I believe `arraycopy` can be faster than a regular for loop. But maybe I'm wrong and it is as fast? I guess we may have to do some benchmarking to find out - but I'm not sure if that is in the scope of this PR.

@jdunkerley what do you think? How much should we spend on this?

I think the gains from the existing safepoints will already be substantial, but indeed we may need to go a bit deeper to ensure that _all_ operations that are bounded by **row count** are interruptible.

I think how big of a priority this is depends on:
1. How much of a pain this still is after this PR,
2. What will be the conclusion of the discussion on alternative and more general solutions (are safepoints 'temporary' and we plan to adapt some more general solution (if at all possible) or do we need to make-do with justsafepoints).